src/management-system-v2/app/(dashboard)/profile/user-data-modal.tsx

```diff
@@ -0,0 +1,155 @@
+'use client';
+
+import { FC, useEffect, useState } from 'react';
+import { Button, Skeleton, Form, Input, Modal, App } from 'antd';
+import { ApiData, ApiRequestBody, useGetAsset, usePutAsset } from '@/lib/fetch-data';
+import { useSession } from 'next-auth/react';
+
+type modalInputField = {
+  userDataField: keyof ApiData<'/users/{id}', 'get'>;
+  submitField: keyof ApiRequestBody<'/users/{id}', 'put'>;
+  label: string;
+  password?: boolean;
+};
+
+type modalInput = ({ password: true } | { password: false; inputFields: modalInputField[] }) & {
+  title: string;
+};
+
+const UserDataModal: FC<{
+  structure: modalInput;
+  modalOpen: boolean;
+  close: () => void;
+  messageApi: ReturnType<typeof App.useApp>['message'];
```

Comment:
Is there a reason to pass in the messageApi instead of doing `const { message: messageApi } = App.useApp();` like in the `user-profile` component?

Reply:
I fixed it in my last push.